smartsim/_core/entrypoints/dragon.py

```diff
@@ -86,28 +84,50 @@ def print_summary(network_interface: str, ip_address: str) -> None:
         )
 
 
-def run(dragon_head_address: str) -> None:
-    global SHUTDOWN_INITIATED  # pylint: disable=global-statement
+def run(
+    dragon_head_address: str, dragon_pid: int, zmq_context: zmq.Context[t.Any]
+) -> None:
     print(f"Opening socket {dragon_head_address}")
-    dragon_head_socket = context.socket(zmq.REP)
+
+    zmq_context.setsockopt(zmq.SNDTIMEO, value=1000)
+    zmq_context.setsockopt(zmq.RCVTIMEO, value=1000)
+    dragon_head_socket = zmq_context.socket(zmq.REP)
     dragon_head_socket.bind(dragon_head_address)
-    dragon_backend = DragonBackend()
+    dragon_backend = DragonBackend(pid=dragon_pid)
 
     server = dragonSockets.as_server(dragon_head_socket)
 
-    while not SHUTDOWN_INITIATED:
-        print(f"Listening to {dragon_head_address}")
-        req = server.recv()
-        print(f"Received request: {req}")
+    print(f"Listening to {dragon_head_address}")
+    while not (dragon_backend.should_shutdown or SHUTDOWN_INITIATED):
+        try:
+            req = server.recv()
+            print(f"Received request {req}", flush=True)
```

Comment:
you added your logger and forgot to use it!

Reply:
Facepalming so hard.